% Table generated by Excel2LaTeX from sheet 'Sheet1'
%
\begingroup
\begin{landscape}
   \tiny
   \noindent
    \begin{longtable}{ l
    				c
				l
				c
				r
				c
				r
				r
				r
				r}
    \caption[Control genomes]{\textbf{Control genomes.} Columns: 1) species name,  2) EMBL-bank accession, 3) phylum, 4) genome size in megabases, 5) number of CDSs annotated in genome, 6) genome G+C content, 7) number of RNIE predictions in genome mode on native sequence, 8) number of RNIE predictions in genome mode on dinucleotide shuffled sequence, 9) number of RNIE predictions in gene mode on native sequence, 10) number of RNIE predictions in gene mode on dinucleotide shuffled sequence.}
    \\
    \toprule
    \textbf{Species} & \textbf{EMBL accession} & \textbf{Phylum} & \textbf{Genome size (MB)} & \textbf{CDSs} & \textbf{G+C content} & \multicolumn{4}{c}{\textbf{Number of predictions}} \\
    & & & & & & \multicolumn{2}{c}{\textbf{Genome}} & \multicolumn{2}{c}{\textbf{Gene}}\\
    & & & & & & \textbf{native} & \textbf{shuffled} & \textbf{native} & \textbf{shuffled}\\
    \midrule
    \textit{Mycobacterium tuberculosis} & AE000516 & Actinobacteria & 4.40 & 4189 & 0.66 & 19 & 0 & 111 & 3\\
    \textit{Streptomyces griseus} & AP009493 & Actinobacteria & 8.55 & 7138 & 0.72 & 72 & 0 & 353 & 2\\
    \textit{Bacteroides thetaiotaomicron} & AE015928 & Bacteroidetes & 6.26 & 4778 & 0.43 & 783 & 2 & 1470 & 44\\
    \textit{Chlamydophila pneumoniae} & AE001363 & Chlamydiae & 1.23 & 1052 & 0.41 & 61 & 3 & 135 & 19\\
    \textit{Prochlorococcus marinus} & AE017126 & Cyanobacteria & 1.75 & 1882 & 0.36 & 81 & 5 & 131 & 22\\
    \textit{Deinococcus radiodurans} & AE000513 & Deinococcus-Thermus & 2.65 & 2579 & 0.67 & 283 & 0 & 506 & 2\\
    \textit{Bacillus subtilis} & AL009126 & Firmicutes & 4.22 & 4245 & 0.44 & 1851 & 4 & 2540 & 54\\
    \textit{Clostridium difficile} & AM180355 & Firmicutes & 4.29 & 3777 & 0.29 & 431 & 8 & 1152 & 58\\
    \textit{Fusobacterium nucleatum} & AE009951 & Fusobacteria & 2.17 & 2067 & 0.27 & 155 & 1 & 457 & 34\\
    \textit{Thermodesulfovibrio yellowstonii} & CP001147 & Nitrospirae & 2.00 & 2033 & 0.34 & 78 & 6 & 176 & 41\\
    \textit{Escherichia coli} & U00096 & Proteobacteria & 4.64 & 4321 & 0.51 & 601 & 6 & 1058 & 35\\
    \textit{Helicobacter pylori} & AE000511 & Proteobacteria & 1.67 & 1566 & 0.39 & 28 & 12 & 128 & 61\\
    \textit{Salmonella enterica} & AE014613 & Proteobacteria & 4.79 & 4323 & 0.52 & 537 & 4 & 980 & 32\\
    \textit{Leptospira interrogans} & AE016823 & Spirochaetes & 4.28 & 3394 & 0.35 & 164 & 18 & 375 & 132\\
    \textit{Ureaplasma parvum} & AF222894 & Tenericutes & 0.75 & 611 & 0.26 & 54 & 0 & 163 & 5\\
    \textit{Fervidobacterium nodosum} & CP000771 & Thermotogae & 1.95 & 1750 & 0.35 & 409 & 3 & 588 & 28\\
    \textit{Methylacidiphilum infernorum} & CP000975 & Verrucomicrobia & 2.29 & 2472 & 0.45 & 50 & 7 & 157 & 52\\
    \bottomrule
    \label{tab:genomes}%
    \end{longtable}%
\end{landscape}%
\endgroup

