% \pagebreak[4]
% \hspace*{1cm}
% \pagebreak[4]
% \hspace*{1cm}
% \pagebreak[4]
%\usepackage[round,colon,authoryear]{natbib}

\chapter{Detecting Rho-independent terminators in genomic sequence with covariance models}
\label{sec:chapterPingpong}
\ifpdf
    \graphicspath{{Chapter4/Chapter4Figs/EPS/}{Chapter4/Chapter4Figs/}}
\fi

\textit{Portions of this chapter are based on the previously published article ``RNIE: genome-wide prediction of bacterial intrinsic terminators'' \parencite{Gardner2011a}. This work is the result of collaboration with Paul P. Gardner (Wellcome Trust Sanger Institute/University of Canterbury). }

\section{Introduction}

Bacteria are thought to utilize two major systems for transcriptional termination: rho-dependent termination, and rho-independent or intrinsic termination \parencite{Peters2011}. Rho-dependent termination relies on a protein cofactor, Rho, a homohexameric ring protein that threads its way along the newly synthesized RNA molecule before causing RNA polymerase (RNAP\nomenclature[Z]{RNAP}{RNA polymerase}) to dissociate at poorly defined pause sites. Intrinsic termination on the other hand, depends primarily on the biophysical characteristics of the sequence being transcribed, and the discovery of these intrinsic terminators in genomic sequence is the subject of this chapter. This chapter will serve largely as background and motivation for the next, in which I develop computational methods for identifying and characterizing transcriptional termination motifs across the bacterial phylogeny.

\subsection{Rho-independent termination}

Intrinsic termination is mediated by short structured RNA motifs known as rho-independent terminators (RITs\nomenclature[Z]{RIT}{Rho-independent terminator}). These are generally characterized by a G+C-rich hairpin followed by a tract of T/U residues. As RNAP transcribes the poly-U tract it pauses, possibly with assistance from the partially formed hairpin structure, allowing full nucleation of the hairpin which melts weak rU-dA bonds within the elongation complex and leads to dissociation of RNAP \parencite{Peters2011}, see figure \ref{fig:rho}. This process is somewhat stochastic, and the probability of successful transcription termination depends on various features of the RIT including stem composition and length, loop composition, length of the poly-U tract, and sequence context of the element \parencite{Larson2008, Cambray2013, Chen2013}. As is well known from the study of transcriptional attenuators and riboswitches \parencite{Henkin2002, Barrick2007, Naville2010}, alternative structures formed upstream of the RIT can also affect termination efficiency, and force exerted on the upstream sequence can increase termination efficiency even in the absence of obvious alternative structures \parencite{Larson2008}.

\begin{figure}[htp]
\begin{center}
\includegraphics[width=14cm]{rhoindy.jpg}
\caption[Rho-independent termination]{\textbf{Rho-independent termination.} A) The RNA polymerase traverses the DNA template strand from $3^\prime$  to $5^\prime$ , synthesizing the nascent RNA molecule. B) As the polymerase nears a termination site, a G+C-rich terminator stem sequence (boxed) is transcribed. C) Formation of a hairpin structure causes the polymerase to pause, and together with a string of unstable rU-dA bonds causes the polymerase to release from the template. Reproduced from \textcite{Gardner2011a}.
} 
\label{fig:rho}
\end{center}
\end{figure}

The degree to which bacteria rely on intrinsic termination varies widely. A bioinformatic analysis examining the computationally predicted minimum free energy (MFE\nomenclature[Z]{MFE}{Minimum free energy}) of gene termini showed that while some species display an enrichment of strong RNA secondary structures at the $3^\prime$ ends of genes, others do not \parencite{Washio1998}. Mutagenesis studies support this conculsion: while Rho is essential in some genomes with fewer apparent intrinsic terminators (for instance, \textit{Salmonella enterica}, see table \ref{tab:core}), it is dispensable in others that are more heavily dependent on intrinsic termination, such as \textit{Bacillus subtilis} \parencite{Quirk1993}. This suggests competition between the two termination systems, leading to clade-specific skews in RIT utilization \parencite{Carafa1990, Kroger1998, Hoon2005}. The accurate prediction of these elements is critical to understanding the regulation of transcript, particularly in light of the $\ge$3000 completed bacterial genomes currently deposited in EMBL-bank. In addition to their obvious role in helping to define operon structures in genomic sequence \parencite{Salgado2013}, they can also be important indicators of cis-RNA regulation \parencite{Henkin2002, Barrick2007, Naville2010}.

\subsection{Previous approaches to identifying intrinsic terminators}

Two main approaches to detecting RITs have been taken over the years: RNA motif descriptors, both expertly constructed \parencite{Lesnik2001} and automatically generated \parencite{Naville2011}; and thermodynamic models of RNA folding to detect hairpins paired with a heuristic scoring scheme for the poly-T tail region \parencite{Ermolaeva2000, Wan2005, Wan2006, Kingsford2007}. Arguably the most popular of these methods has been TransTermHP \parencite{Kingsford2007}, an example of the second approach.

\begin{figure}[htp]
\begin{center}
\includegraphics[width=14cm]{transterm}
\caption[TransTermHP motif]{\textbf{TransTermHP motif.} Schematic of the terminator motif that TransTermHP searches for. The terminators consist of a short stem-loop hairpin followed by a thymine-rich region on their $3^\prime$ side. TransTermHP is generally restricted to find terminators where each side of the stem is $\ge$4 nt, the length of the loop is $\ge$3 nt and $\le$13 nt, and the total length of the stem-loop is $\le$59 nt. Reproduced under a Creative Commons Attribution License (CCAL) from \textcite{Kingsford2007}.
} 
\label{fig:transterm}
\end{center}
\end{figure}

The TransTermHP algorithm takes a windowed approach to detecting RITs (figure \ref{fig:transterm}), in common with other thermodynamic approaches. In order to avoid the computational cost of predicting local secondary across the entire genome, TransTermHP first scans overlapping windows of 6 bases for those containing at least 3 T residues. Upon finding such a window, TransTermHP performs a dynamic programming procedure to predict potential hairpin structures, using a simplified version of the Zuker algorithm \parencite{Zuker1981} for {\it in silico} RNA folding parameterized using a set of experimentally validated {\it Escherichia coli} RITs \parencite{Ermolaeva2000}. This is then combined with a heuristic score for the quality of the poly-T tail \parencite{Carafa1990} which rewards T residues occurring closer to the closing base-pairs of the predicted hairpin structure. Candidate RITs are then filtered on stem length, loop length, and total length (see the caption of figure \ref{fig:transterm} for details). Finally, the combined score of surviving candidates is compared to the scores of predicted terminators in random sequence with similar GC content to that of the target genome to provide a measure of prediction quality. Search is apparently also limited to regions surrounding stop codons (\cite{Kingsford2007}; see also the discussion of the beta benchmark below), though the exact boundaries on the search space are not explicitly given in the TransTermHP documentation or publication.

This methodology presents a number of problems. First, while the thermodynamic method used to predict hairpin structures likely places some implicit restrictions on the sequence composition of the hairpin structure, it does not explicitly model conservation of residue composition across terminators. Conservation of residue composition could arise due to convergent evolution of terminator structures under selection for properties that promote strong termination in the host species, or as the results of homology between RITs due to their descent from a common ancestor deposited by transposable elements, as has previously been hypothesized \parencite{Naville2010}. In addition, windowed searching for and heuristic scoring of the poly-T tail is unlikely to accurately capture the true constraints on this feature. We show here that explicitly modelling residue conservation improves detection of RITs. Secondly, the comparison to random sequence with similar GC content is unlikely to be an adequate control: it has been shown that considering dinucleotide content of sequences is critical to determining the significance of their secondary structure \parencite{Workman1999}. Though the method of generating random sequence is not explicitly stated in \textcite{Kingsford2007}, it seems unlikely that it was the product of dinucleotide shuffled sequence or a first-order Markov chain, as would be required to preserve dinucleotide frequencies. In fact TransTermHP does not appear to consider base-stacking effects in its predictions whatsoever. Finally, restricting search to the regions around annotated gene termini, or rewarding candidate RITs for being in these regions, is both somewhat artificial and requires accurate gene annotation, which remains a challenge. 

\subsection{Covariance models}

Our method overcomes many of the problems in previous RIT detection methods through the use of {\it covariance models} (CMs\nomenclature[Z]{CM}{Covariance model}), a special case of stochastic context-free grammars \parencite{Eddy1994, Sakakibara1994}. CMs are sophisticated statistical models which incorporate information about both sequence and secondary structure conservation. They are perhaps most easily understood through analogy to the simpler profile hidden Markov models (HMMs) \parencite{Eddy1998}. A typical method for HMM construction takes as its input an alignment of sequences. For each column of the alignment, a {\it node} is constructed, consisting of three {\it states}: match, insert, and delete. The match state models the residue distribution at the alignment position, while the insert and delete states model the probability and length distributions of insertions and deletions beginning at this column, respectively. The mathematics of HMMs have been well explored, and efficient dynamic programming algorithms exist for training (the Baum-Welch algorithm), assigning a probability to a sequence being produced by the model (the Forward algorithm), and finding the most probable parsing of a sequence (the Viterbi algorithm).

\begin{figure}[htp]
\begin{center}
\includegraphics[width=14cm]{cm}
\caption[Covariance model architecture]{\textbf{Covariance model architecture.} A) A toy multiple alignment of three RNA sequences, with 28 total columns, 24 of which will be modeled as consensus positions. The [structure] line annotates the consensus secondary structure: angle brackets mark base pairs, colons mark consensus single-stranded positions, and periods mark “insert” columns that will not be considered part of the consensus model because more than half the sequences in these columns contain gaps. B) The structure of one sequence from A, the same structure with positions numbered according to alignment columns, and the guide tree of nodes corresponding to that structure, with alignment column indices assigned to nodes (for example, node 5, a MATP match-pair node, will model the consensus base pair between columns 4 and 14). C) The state topology of three selected nodes of the CM, for two MATP nodes and one consensus “leftwise” single residue bulge node (MATL, “match-left”). The consensus pair and singlet states (two MPs and one ML) are white, and the insertion/deletion states are gray. State transitions are indicated by arrows. Reproduced under a Creative Commons Attribution License (CCAL) from \textcite{Nawrocki2007}.
} 
\label{fig:cm}
\end{center}
\end{figure}

CMs are similar to profile HMMs, with the extension that they can additionally model dependence between alignment positions (see figure \ref{fig:cm}); rather than nodes being constructed from alignment columns, they are built from structural elements, i.e. pairing bases, annotated in the alignment (figure \ref{fig:cm}B). This increases the complexity of node architecture (figure \ref{fig:cm}C), as each node must now contain states to match both bases in a pair, either one of a pair individually if its partner has been deleted, insertions on either side of the pair, and base pair deletions. Analogs to the Baum-Welch, Forward, and Viterbi algorithms exist for CMs: expectation-maximization using the inside-outside algorithm, the inside algorithm, and the Cocke-Younger-Kasami (CYK\nomenclature[Z]{CYK}{Cocke-Younger-Kasami (algorithm)}) algorithm, respectively. Unfortunately, modeling the dependence between positions, that is moving from a regular grammar such as an HMM to a context-free grammar such as a CM, comes at a considerable computational cost due to the restrictions imposed by the Chomsky hierarchy \parencite{Chomsky1959}, roughly equivalent to adding an additional dimension to the dynamic programming matrix. In this study we have used the Infernal package \parencite{Nawrocki2009}, which implements CMs and associated algorithms for RNA sequence analysis, and includes a number of heuristics for increasing the speed of CM-based searches including adaptive banding of the dynamic programming matrix \parencite{Nawrocki2007} and HMM pre-filters based on HMMER 3 \parencite{Eddy2011}. Importantly, Infernal also incorporates a null model for scoring sequence hits; for sequence that matches the CM, the probability of this match is compared to the probability of a match to the null model. This comparison is expressed as a $log_2$ odds ratio, or bitscore, and from this further statistics, such as an expect value (E-value\nomenclature[Z]{E-value}{Expect value}), can be calculated. Covariance models are widely used in RNA homology search, most notably by the Rfam database \parencite{Burge2013} and the tRNAscan-SE tool \parencite{Lowe1997} for predicting tRNAs in genomic sequence.

\section{Methods}
\textit{Paul P. Gardner implemented and benchmarked the RNIE tool. Eric P. Nawrocki (Howard Hughes Medical Institute Janelia Farm Research Campus) assisted in optimizing Infernal parameters for RIT search. Zasha Weinberg (Howard Hughes Medical Institute/Yale University) provided the Rnall-Breaker algorithm and advice. I designed and implemented the analysis which lead to the discovery of the putative mycobacterial termination motif.}

\subsection{Construction of a covariance model for Rho-independent terminators}

171 and 891 experimentally validated RIT sequences from \textit{Escherichia coli} and \textit{Bacillus subtilis}, respectively, were collected from the \textit{E. coli} Database Collection (ECDC; \cite{Wahl1995}) and the supplementary information of \textcite{Hoon2005} and manually curated based on evidence quality, leaving a set of 981 RIT sequences. These sequences were subjected to iterative rounds of alignment, structure prediction, homology search and refinement. Alignments and secondary structures were predicted using WAR \parencite{Torarinsson2008}, CMfinder \parencite{Yao2006}, and MLocarna \parencite{Will2007}, iteratively refined using Infernal \parencite{Nawrocki2009}, then manually refined using the RALEE emacs environment \parencite{Griffiths-Jones2005}. Sequence searches were performed using the resulting CM against EMBL using the Rfam pipeline \parencite{Gardner2009}, and additional sequences were incorporated in to the alignment based on the following criteria: i) the maximum similarity to an existing seed sequence had to be 95\% and the minimum 60\%, ii) the minimum fraction of canonical base pairs had to be 75\%, iii) the sequence annotation should not contain terms like contaminant, pseudogene, repeat or transposon and iv) they must score above a bitscore threshold of 20. These sequences were then manually checked for their position near a gene terminus. This resulted in 1117 aligned sequences, which were further split in to two groups based on how well they matched the resulting CM. Those scoring with a bitscore over 14 were placed in alignment A, those scoring less were placed in alignment B. These alignments were then again automatically refined using Infernal before a final round of manual refinement. 

\subsection{RNIE run modes}

As described in the introduction, algorithms for performing inference with CMs and be very slow, and as a result Infernal implements a number of filters to reduce the number of sequences which proceed to a full CM-based homology search. In response to this, two modes for RNIE were developed: genome mode meant for large-scale searches, which enables HMM filters and adaptive banding and uses the CYK algorithm with a higher threshold for reported RIT predictions; and gene mode meant annotation of relatively short sequence regions, which disables Infernal's filtering mechanisms and uses the Inside algorithm with a lower threshold for reporting RIT predictions. Genome mode scans sequence at $\sim$43 kb/s\nomenclature[Z]{kb}{Kilobase} with a low false positive rate of $\sim$1.7 FP/Mb\nomenclature[Z]{Mb}{Megabase}. The sensitivity, positive predictive value and Matthews' correlation coefficient for this mode is 0.70, 0.79 and 0.74. Gene mode scans at $\sim$1kb/s, and the false positive rate, positive predictive value and Matthews' correlation coefficient are $\sim$9.6 FP/Mb, 0.45 and 0.61, respectively. The Infernal parameters used for genome and gene mode, respectively, are

\begin{quote}
\begin{verbatim}
cmsearch -T 16 -g --fil-no-qdb --fil-T-hmm 2 
--cyk --beta 0.05 CM query_sequence.fasta

cmsearch -T 14 -g --fil-no-qdb --fil-no-hmm 
--no-qdb --inside CM query_sequence.fasta
\end{verbatim}
\end{quote}

\subsection{Definitions}

For the purposes of benchmarking, the following measures were used \[Sensitivity = \frac{TP}{TP + FN} \] \[ PPV = \frac{TP}{TP + FP} \] \[ FPR = \frac{FP}{\text{Total length in kb}} \] \[ MCC = \frac{TP \times TN - FP \times FN}{\sqrt{(TP + FP)(TP + FN)(TN + FP)(TN + FN)}} \] where any prediction that covered a known RIT by at least one nucleotide was considered a true positive (TP), any prediction that did not overlap a known RIT was considered a false positive (FP), a missed RIT was considered a false positive (FP), and the number of unclassified, non-RIT sequence were considered true negatives (TN).

\section{Results}
\subsection{Alpha benchmark}

\begin{figure}[htp]
\begin{center}
\includegraphics[width=14cm]{alpha.jpg}
\caption[Alpha benchmark]{\textbf{Alpha benchmark.} The accuracy of RNIE compared to existing methods of terminator prediction. The left figure shows a ROC plot for four independent methods. The middle figure compares the sensitivity and PPV for the four methods. The figure on the right shows the speeds for each algorithm in kilobases per second. Reproduced from \textcite{Gardner2011a}.
} 
\label{fig:alpha}
\end{center}
\end{figure}

\subsection{Beta benchmark}

\input{Chapter4/Chapter4Tables/genomes.tex}

\begin{figure}[htp]
\begin{center}
\includegraphics[width=14cm]{beta.jpg}
\caption[Beta benchmark]{\textbf{Beta benchmark.} Ideal terminator predictors will generally produce predictions that are immediately $3^\prime$ to annotated genes on native sequence and no predictions on shuffled controls. For all the test genomes in Table 1 (excluding {\it E. coli} and {\it B. subtilis}), we computed the distance to the nearest $3^\prime$ genic element, including CDSs, ncRNAs and riboswitches. This was done for both native sequences and dinucleotide shuffled control sequences with corresponding gene annotation transferred to the controls. The figure on the left shows the distribution of distances for RNIE genome and gene modes and for the TransTermHP method. Inset is a barplot showing the total number of predictions for each method on native and shuffled genomes. The figures on the right show the percentage of genes that have a predicted RIT in the region $-50$ to $+150$ from an annotated $3^\prime$-end of a CDS or ncRNA across all the genome sequences described in Table 1. The upper panel illustrates the results for the native genomes, while the lower panel illustrates results for the permuted genomes. Reproduced from \textcite{Gardner2011a}.
} 
\label{fig:beta}
\end{center}
\end{figure}

\subsection{A novel termination motif in \textit{Mycobacterium tuberculosis}}

\begin{figure}[htp]
\begin{center}
\includegraphics[width=14cm]{myco.jpg}
\caption[Putative mycobacterial transcription termination motif]{\textbf{Putative mycobacterial transcription termination motif.} A) The frequency of TRITs and RITs near the terminal regions of {\it M. tuberculosis} (EMBL accession: AE000516) genic features. B) The distribution of structural stability derived P-values for the most significant {\it M. tuberculosis} terminal regions coloured by TRIT (red), RIT (black) or unclassified (blue). C) The secondary structure and sequence conservation of the TRIT motif as displayed by R2R. (D\&E) Sequence logos generated for the $5^\prime$ D) and $3^\prime$ E) halves of an alignment of the 147 copies of TRIT in the {\it M. tuberculosis} genome. Reproduced from \textcite{Gardner2011a}.
} 
\label{fig:myco}
\end{center}
\end{figure}
