%%% Thesis Introduction --------------------------------------------------
\chapter{Introduction}
\markboth{Introduction}{Introduction}
\ifpdf
    \graphicspath{{Introduction/IntroductionFigs/PNG/}{Introduction/IntroductionFigs/PDF/}{Introduction/IntroductionFigs/}}
\else
    \graphicspath{{Introduction/IntroductionFigs/EPS/}{Introduction/IntroductionFigs/}}
\fi

Bacteria possess a remarkable ability to adapt. This ability has allowed bacteria to colonize almost every environment on Earth, from deep sea hydrothermal vents \parencite{Jorgensen1992} to cryogenic brine lakes \parencite{Murray2012} to animal hosts \parencite{Finlay1997}. Indeed, the ability of bacteria to establish symbiotic relationships with host cells was a critical step in the origin of so-called ``higher" eukaryotic life \parencite{Sagan1967}. While the origins of some bacterial adaptations are buried in the deep time of over 1.5 billion years of evolution \parencite{Doolittle1996}, such as the differing bauplans observed across phyla, others are far more recent, such as the emergence of \textit{Yersinia pestis} as a human pathogen around 20,000 years ago \parencite{Achtman1999} or the contemporary development of specialized invasive lineages of non-typhoidal \textit{Salmonella} in immunocompromised individuals in sub-Saharan Africa \parencite{Feasey2012, Okoro2012}. Many factors likely contribute to this continuous adaptation, including large population sizes, short generation times, wide-spread homologous recombination between related strains, and a capacity for horizontal gene transfer. These factors, particularly homologous recombination and horizontal gene transfer, make the definition of species in bacteria contentious \parencite{Achtman2008, Doolittle2009}, and have led to some questioning the viability of a bacterial species concept altogether. For the present I will leave these matters to those better informed than myself, and work within the established, though flawed, taxonomy.

The work in this thesis is concerned with the study of bacterial evolution and adaptation on two very different time scales. In the first section, consisting of chapters 1, 2, and 3, I describe a recently emerged high-throughput technology for probing gene function, transposon-insertion sequencing \parencite{Barquist2013}, and its application to the study of functional differences in two important human pathogens, \textit{Salmonella enterica} subspecies \textit{enterica} serovars Typhi and Typhimurium. These two serovars diverged only approximately 50,000 years ago \parencite{Kidgell2002}, yet have developed very different host ranges and cause very different diseases, with \textit{S.} Typhi causing a life-threatening system disease exclusively in humans, and \textit{S.} Typhi causing primarily a mild gastrointestinal disease in a wide range of hosts. Chapter 2 uses transposon-insertion sequencing to probe differences in gene requirements during growth on rich laboratory media, revealing differences in requirements for genes involved in iron-utilization and cell-surface structure biogenesis, as well as in requirements for non-coding RNA \parencite{Barquist2013a}. In chapter 3 I more directly probe the genomic features responsible for differences in serovar pathogenicity by analyzing transposon-insertion sequencing data produced following a two hour infection of human macrophage, revealing large differences in the selective pressures felt by these two closely related strains in the same environment.

The second section, chapters 4 and 5, uses statistical models of sequence variation, i.e. covariance models, to examine the evolution of intrinsic termination across the bacterial kingdom. Chapter 4 provides background and motivation in the form of a method for identifying Rho-independent terminators using covariance models built from deep alignments of experimentally-verified terminators from \textit{Escherichia coli} and \textit{Bacillus subtilis} \parencite{Gardner2011a}. In the course of the development of this method I discovered a novel putative intrinsic terminator in \textit{Mycobacterium tuberculosis}. In chapter 5, I extend this approach to \textit{de novo} discovery of intrinsic termination motifs across the bacterial phylogeny. I present evidence for lineage-specific variations in canonical Rho-independent terminator composition, as well as discover seven non-canonical putative termination motifs. Using a collection of publicly available RNA-seq datasets, I provide evidence for the function of these elements as \textit{bona fide} transcriptional attenuators.