%\usepackage[round,colon,authoryear]{natbib}

\chapter{Querying bacterial genomes with transposon-insertion sequencing}
\label{sec:chapter_piRNAs}
\ifpdf
    \graphicspath{{Chapter1/Chapter1Figs/PNG/}{Chapter1Chapter1Figs/PDF/}{Chapter1/Chapter1Figs/}}
\else
    \graphicspath{{Chapter1/Chapter1Figs/EPS/}{Chapter1/Chapter1Figs/}}
\fi

%To - do 
%1. Pie chart of genomic properties, ratio of unique to non unique mappers, length distribution
%See Lai paper

% conversion of the references with: find \((\d+)\) replace \\citep\{\1\}

\section{Introduction}

The introductory chapter discussed the establishment of the field of non-coding RNAs and the underlying technologies that were crucial for its development. One type of non-coding RNAs (ncRNAs) has also been discovered only recently through advances in second generation sequencing, the PIWI-interacting RNAs (piRNAs) (see Chapter \ref{piRNAs}). The piRNAs are associated with germ-cells across the animal kingdom, where they mostly have the conserved function in protection against transposable elements. However, piRNAs in mammals come in two flavours; the embryonic pre-pachytene piRNAs that are the genome's immune system against the transposable elements (TEs) and the adult pachytene piRNAs whose function remains unknown. The biogenesis and evolution of pachytene piRNAs could potentially explain their function and are the main topic of the next chapter. 

\newpage
\subsection{Pachytene piRNAs}
The mammalian pachytene piRNAs are found in great abundance in late spermatogenesis \citep{Meikar:2010jm}. However, they do not map to repeats or any known sequences and are therefore likely to have a role outside of TE regulation. 

The pachytene piRNAs are similar to their embryonic counterparts in their sequence heterogeneity, but come from different genomic loci. First, pachytene piRNAs also map to genome in clusters, but they show little overlap with pre-pachytene piRNAs  \citep{Aravin:2007hw}. The pachytene clusters are fewer in number, but more abundant in piRNAs. They also contain a higher proportion of uniquely mapping reads (80\% vs 60\%) and are depleted of transposable elements \citep{Betel:2007p580}. Finally, the pachytene clusters are often associated with $3\textprime$ UTRs of coding regions \citep{Robine:2009p378, Gan:2011in}, although the significance of this observation has not yet been explained. 

The current model of pachytene piRNA biogenesis presumes production of primary piRNA transcripts, recognised by unknown endo- and exonucleases, which prime the Piwi proteins \citep{Siomi:2011gh}. However, the putative primary transcripts from piRNA cluster loci are not distinguished by their secondary structure or their sequence composition \citep{Betel:2007p580}. Furthermore, they do not have the repeat-binding antisense partners to promote their amplification, as in the embryonic piRNAs (Chapter \ref{sec:chapterPingpong}, Figure \ref{fig:pingpong}). The long non-coding primary transcript that gives rise to piRNAs have been observed to sometimes have a complementary antisense RNA. Such a complementary transcript might have eluded conclusive characterisation due to its instability \citep{Watanabe:2006ij}.

Although many features of piRNA maturation have been identified in individual species, the evolution of piRNA loci is less explored. In closely related mammals piRNA production is restricted to syntenic genomic locations, while the piRNA sequences themselves are poorly conserved \citep{Girard:2006gu}. Mammals have maintained certain genomic properties of piRNA producing loci. For example, the piRNAs are highly expressed from a small number of genomic loci across multiple species \citep{Murchison:2008kp}. Identification of these loci in other species based on homology prediction is difficult, since gene expression in testis is evolutionarily accelerated \citep{Chan:2009iq, Brawand:2011du}. 

Currently, there are no studies of the production of piRNAs at the level of transcription machinery, epigenetic marks and transcriptional regulation that could answer the question of biological specificity of primary piRNA transcripts. 

\subsection{RNA polymerases}

The transfer of information from DNA to RNA is a highly regulated process that involves an interplay between the DNA molecule, a protein complex and the nascent RNA molecule. In eukaryotes, transcription is performed by five DNA-dependent RNA polymerases (RNAP, Pol), \nomenclature{RNAP}{DNA-dependent  RNA polymerase}  each acting on a different class of genes. Plants lack the piRNA machinery and their polymerases IV and V are instead involved in TE silencing through RNA-directed DNA methylation \citep{Haag:2011cj}. Three RNA polymerases present in animals are the potential candidates for transcription of piRNA cluster loci: DNA dependent RNA polymerases I, II and III.

RNA polymerase I \nomenclature{Pol}{RNA polymerase} 
 is a highly processive protein complex responsible for over 75\% of transcripts in the cell \citep{Russell:2005p585}. This polymerase mainly transcribes the long genomic regions of ribosomal 45S RNA, which matures into 18S, 5.8S and 28S rRNAs. The rRNA transcription rate is determined by the tissue specific activity of the nucleolar transcription factor UBF, while the number of actively used genomic loci plays a less significant role \citep{Stefanovsky:2006p587}. Furthermore, the pool of active ribosomal genes changes during differentiation according to the cell specific modifications in ribosomal DNA methylation \citep{Sanij:2008p588}. In conclusion, the processivity, the length of transcripts and the cell specific transcription support the hypothesis that RNA polymerase I could be the main part of the piRNA transcription machinery.

RNA polymerase II  transcribes only 10\% of cellular transcripts \citep{Werner:2009p589}, the majority of which are the protein-coding genes. However, the variability in RNA processing partners allows this polymerase to produce both the long polyadenylated mRNA and ncRNA transcripts without the polyadenylated (poly(A))\nomenclature{poly(A)}{polyadenylated} tail \citep{Zhang:2012p590}. Therefore, RNAP II is also able to transcribe a variety of non-coding RNAs of different lengths: from miRNAs precursors of 100bp, longer snRNAs and snoRNAs, to lncRNAs of several hundred kilobases (kb) \nomenclature{kb}{kilo base} \citep{Kuehner:2011p591}. Due to its versatile nature and ubiquity in RNA transcription, RNAP II presents the most likely candidate for the transcription of piRNA genes.

Polymerase III was historically mainly associated with translation due to the transcription of aminoacyl tRNAs and 5S rRNAs \citep{Dieci:2007p592}. Recently, a multitude of RNAP III-produced non-coding RNAs were described in a variety of processes from RNAP II regulation and splicing to RNA processing and multi-drug resistance \citep{White:2011p593}. Besides the involvement in production of ncRNA, this polymerase is also specific for its processing of repeat elements such as SINEs \citep{Kramerov:1985p594}. The association with various non-coding RNAs suggested a possible role of RNAP III in transcription of piRNA loci.

In summary, the transcription of piRNA clusters could be performed by any of the previously described RNA polymerases due to the variety of their substrates or the length of their products. However, in addition to the transcription machinery, various epigenetic marks have to be assessed to describe the properties of transcription of piRNA genomic loci. 

\subsection{The epigenetic landscape}

Nuclear genetic material in the form of chromatin contains a level of information beyond the nucleotide composition. The information is stored in the chemical changes to nucleotides themselves or in the composition and positioning of the DNA-binding proteins \citep{Portela:2010p595}. The first class of these epigenetic modifications includes the methylation of a Cytosine nucleotide from Cytosine-Guanine dinucleotide repeats (CpG islands) and has been related to silencing of the promoter regions \citep[reviewed in][]{Jin:2011p596}. The second includes various modifications of histones known as the ``histone code'', which have different functions in chromosome stability, translation and transcription \citep{Turner:2012p597}. The epigenetic landscape of piRNA transcripts was monitored on three major marks of transcription located of the H3 histone: trimethylation of the fourth lysine (H3K4me3), trimethylation of the 36th lysine (H3K36me3) and trimethylation of the 27th lysine (H3K27me3). 

The three major histone marks H3K4me3, H3K36me3 and H3K27me3 denote the transcription initiation, elongation and inhibition, respectively. The presence of the H3K4me3 histone modification marks the initiation sites of the loci that are actively transcribed by RNAP II \citep{SantosRosa:2002p598}, although the association with RNAP III promoters has also been reported \citep{White:2011p593}. Binding of this polymerase to the promoter initiation complex recruits the Set1 methylase that actively modifies the H3K4 site \citep{Hon:2009p601}. The trimethylated H3K4 is thereafter used as an anchor for other transcription initiation proteins, promoting further RNAP II binding to the initiation locus \citep{Vermeulen:2007p600} as well as defining recombination hotspots \citep{Oliver:2009p602}. The same modification of the 36th lysine on the H3 histone (H3K36me3) marks the active elongation \citep{KolasinskaZwierz:2009p603}. Detachment of the polymerase from the promoter sites promotes binding of the Set2 protein that actively methylates the 36th lysine \citep{Hon:2009p601}. The enrichment of this mark varies between the introns and the exons on an evolutionary conserved level, involving this modification in both transcription and splicing processes \citep{KolasinskaZwierz:2009p603}. Finally, the H3K27me3 marks transcript inhibition through association with the condensed and inactivated chromatin form, also termed ``heterochromatin'' \citep{Ringrose:2004p604}. In the key developmental genes it is present at the promoter sites in combination with H3K4me3, whereby marking the ``bivalent genes'' that are inactivated and quick to respond to stimuli \citep{Bernstein:2006p605}. This mark is deposited by the silencing Polycomb complex in combination with H3K9me3 \citep{Sawarkar:2010p606} and can be used to test whether piRNA loci are specifically inhibited in the non-germ-line tissues.

\subsection{Outline}


In order to determine the piRNA-generating properties of piRNA cluster loci, an extensive assessment of their genomic and transcriptomic landscape was performed. Information on RNA from genomic piRNA loci was combined with publicly available piRNA datasets  to map the location and activity of postnatal pachytene piRNA clusters. Multiple epigenetic marks were also tested in testes of mouse as well as in  \textit{C. familiaris} and \textit{M. domestica}, two additional divergent mammals. The experimental part of the work was performed by Claudia Kutter from the Duncan Odom laboratory at Cancer Research Institute, Cambridge UK, while I performed the computational analysis. 

\section{Results}
	 
	\subsection{Genomic loci of primary piRNA transcripts}
  
In order to investigate the properties of primary piRNA precursors, a set of high confidence, intergenic piRNA clusters had to be created with a minimal amount of possible transcription degradation products. This was achieved in three steps: definition and filtering of piRNAs from annotated regions, identification of active transcripts from the RNA-seq analysis that overlap with the piRNA loci and confirmation of active transcription through ChIP-seq analysis of H3K4 trimethylation.

\begin{figure}[hbtp!]
\begin{center}
\includegraphics[width=14cm]{pipeline}
\caption[Identification of piRNA precursors]{Identification of piRNA precursors. A. Pie chart depicts the categories of genomic annotation of piRNA reads. The piRNAs were retrieved from publicly available sequencing data for adult \citep{Robine:2009p378}, or embryonic testis \citep{DeFazio:2011dua}. The reads were annotated with ENSEMBL annotation. The main categories represent intergenic, repeat associated intergenic (LTRs, SINEs, and LINEs), and genic loci. The genic loci are further separated into non-coding RNA (ncRNA), protein-coding, $5\textprime$ UTR and $3\textprime$ UTR regions. The piRNAs from intergenic regions that were not annotated (purple) were used in the next steps. B. Flow diagram illustrates the assembly of testis-expressed piRNAs and their precursor transcripts marked by H3K4me3 at their transcriptional initiation. Transcripts from mouse testis tissue were identified with RNA-seq while the DNA-binding sites of H3K4me3 were assessed with ChIP-seq.  The piRNA precursors are classified based on directionality of transcription into bidirectional and unidirectional. Numbers (top to bottom) indicate the count of genomic piRNA loci, smRNA sequencing reads mapping to these loci and the average of piRNAs per one kilobase of DNA.
} 
 \label{fig:pipeline}
\end{center}
\end{figure}

Publicly available piRNA sequencing data were used that contained sequence reads from both immunoprecipitated Piwi proteins expressed in the adult mouse testis, Mili and Miwi (Methods, Section \ref{sec:GEO.piRNA.libraries}) \citep{Robine:2009hy}. The sequence reads were combined and aligned to the mouse genome (Methods, Section \ref{sec:genome.mapping.new}). As expected, when compared to the embryonic testis piRNAs from Mili immunoprecipitations \citep{DeFazio:2011dua}, a larger percentage of reads mapped to unannotated intergenic regions with the proportion of uniquely mapping reads 85\% in adult testis compared to 54\% in the embryo (Figure \ref{fig:pipeline}).



\begin{figure}[hbtp!]
\begin{center}
\includegraphics[width=13cm]{bidirectional_unidirectional}
\caption[The majority of piRNAs derive from bidirectional transcription.]{The majority of piRNAs derive from bidirectional transcription. Examples of piRNAs mapping to A. bidirectional (n=42) or B. unidirectional (n=61) piRNA loci in mouse. Small RNA sequences (yellow) and total RNA sequences (red) are shown. The y-axis represents normalised read density (plus, sense; minus, antisense). The x-axis shows the chromosomal location of each example. C. The number of bidirectional and unidirectional loci (rasterised red) and individual piRNAs (rasterised yellow) mapping to each class. D. The 30 most transcribed bidirectional and unidirectional piRNA clusters are displayed in decreasing order of piRNA abundance. Expression of total RNA (red) and piRNAs (yellow) is illustrated as adjacent bars for each locus; the y-axis indicates strandedness (plus, sense; minus, antisense).} 
 \label{fig:bidirectional_unidirectional}
\end{center}
\end{figure}


The clusters were called with the threshold of 50 piRNAs per 1 kb window (Methods, Figure \ref{fig:cluster_cutoffs}) and the sequences were cleaned from any possible transcripts from annotated regions (Methods, Section \ref{sec:cluster.cleaning}). This lead to the identification of 178 intergenic piRNA loci, 44 bidirectional and 134 unidirectional, together containing 78\% of total reads. The majority of piRNAs (81.5\%) originated from transcripts that map to bidirectional loci, with about 1022 piRNAs per 1kb, while the remaining 18.5\% mapped to the unidirectional ones, with 714 reads mapped per kb of locus length (Figure \ref{fig:pipeline}). 

After the detection of piRNA loci of interest, the next step was to annotate them with transcript information, as well as the marks of active initiation sites. Therefore, the analysis of total RNA sequencing (RNA-seq) was performed on the adult testis and liver transcriptome (Methods, Section \ref{sec:RNA.seq.libraries}). After the computational assembly of RNA-seq sequences (Methods, Section \ref{sec:cufflinks}), transcript models for piRNA precursors were built by overlaying obtained RNA transcripts with identified piRNA clusters and maintaining directionality of transcription (Figure \ref{fig:bidirectional_unidirectional}). Furthermore, data from  ChIP-seq on H3K4me3 (Methods, Section \ref{sec:ChIPseq.libraries}) were analysed and overlaid with the peaks of enriched DNA binding  (Methods, Section \ref{sec:peak.calling}) and the piRNA transcript models in order to define precursors with independent transcription initiation sites. In that way 42 bidirectional and 61 unidirectional piRNA clusters were identified, with 80.6 and 14.3 percent of total piRNAs, respectively. The expression of these filtered piRNA primary transcripts was comparable to protein-coding genes (Figure \ref{fig:expression_comparison}). The total piRNA clusters, and separate bidirectional and unidirectional clusters did not differ significantly from the coding regions, with the Wilcoxon test p-values of 0.65, 0.33 and 0.87, respectively. These results indicate that unlike microRNA primary transcripts \citep{Kirigin:2012cl}, piRNA precursors are detectable despite downstream processing. 


\begin{figure}[hbtp!]
\begin{center}
\includegraphics[width=14.0 cm]{expression_comparison}
\caption[Expression levels of piRNA transcripts]{Expression levels of piRNA transcripts are similar to exons of protein-coding genes. Box plots show normalised expression levels in RPKM (reads per kilobase of annotated region, per million of library reads) based on total RNA-seq from mouse testis (y-axis) measured over the entire protein-coding gene loci, which includes $5\textprime$ UTR, exons, introns and $3\textprime$ UTR; coding sequence of protein-coding genes; piRNA producing transcripts, further categorised into bidirectional and unidirectional precursors.
} 
 \label{fig:expression_comparison}
\end{center}
\end{figure}


In conclusion, the intergenic piRNAs derived from few distinct genomic locations were mostly produced from the bidirectional transcripts and mostly arose from loci with independent transcription initiation sites (Appendix, Table \ref{tab:mouseClusters}). 


	\subsection{Characterisation of primary piRNA transcripts}

The primary piRNA transcripts were so far poorly characterised in the literature. Little was known on RNA transcription machinery, polyadenylation, coding potential, as well as the associated epigenetic marks.

After the definition of genomic loci of primary piRNAs I wanted to assess which RNA polymerase is involved in their biogenesis. Thus, the data from ChIP-seq experiments in mouse testis and liver were analysed for all three mammalian polymerases (Methods, Section \ref{sec:ChIPseq.libraries}). The selected piRNA clusters were occupied by RNA polymerase II (Figure \ref{fig:polymerases}, A). When the sequences for each of the polymerases were annotated to the genome (Methods, Section \ref{sec:genome.annotation}), polymerase I bound the 45S subunits, polymerase II protein-coding genes and mRNAs, while polymerase III occupied the regions of 5S rRNA and tRNA, with the fold change against background (input) greater than two and p-value smaller than 0.01(Figure \ref{fig:polymerases}, B). The smaller intensity and significance of the signal for the testis samples was likely due to the greater homogeneity of the liver tissue. 

\begin{figure}[hbtp!]
\begin{center}
\includegraphics[width=14cm]{polymerases}
\caption[RNA polymerase II transcribes piRNA genes.]{RNA polymerase II transcribes piRNA genes. A. A bidirectional piRNA gene locus annotated with reads of piRNAs (yellow) totalRNA (red), and occupancy of Pol I, Pol II, and Pol III (black). The y-axis specifies normalised read densities (plus, sense; minus, antisense). The stronger signal for piRNAs, as depicted in the higher RPKM levels is caused by the immunoprecipitation step that reduces the background. B. Differential enrichment versus input of all three RNA polymerases over annotated gene loci in testis and liver. Significance is indicated by circle radius (high: large; low: small). Log (base 2) fold-changes over input are shown from blue (negative) to blue (positive).} 
 \label{fig:polymerases}
\end{center}
\end{figure}


\begin{figure}[hbtp!]
\begin{center}
\includegraphics[width=14cm]{precursor_properties}
\caption[Properties of piRNA precursors]{Properties of piRNA precursors. A. Polyadenylation frequency of piRNA precursor transcripts. Ranking of piRNA precursor transcripts based on occurrence in either the random (left) or oligo(dT) (right) primed reverse transcription followed by quantitative PCR normalised against Actin B. Each row represents a piRNA precursor, Mili and Dicer (Dcr) as control for polyadenylated transcripts and Histone H1 as control for non-polyadenylated transcripts. The x-axis displays the percentage of each transcript present either in the random or oligo(dT) primed reverse transcription reaction. Figure adopted from Claudia Kutter, with permission. B. Coding potentials of primary piRNA transcripts. Coding potential was calculated for 100 testis-expressed protein-coding genes, bi- and unidirectional piRNA precursor transcripts identified in this study, as well as randomly assembled sequences of bi- and unidirectional piRNA precursor transcripts. The y-axis shows the coding potential score (the higher the score the more likely that an open reading frame and functional protein domains were identified). Asterisks indicate that the coding potential of bi- and unidirectional piRNA precursor transcripts as well as the randomly assembled sequences of bi- and unidirectional piRNA precursor transcripts were significantly lower compared to the protein-coding genes (Wilcoxon test, bidirectional p-value=8.6x10\textsuperscript{-09}, unidirectional p-value=5.2x10\textsuperscript{-12} as well as random bidirectional p-value=7.6x10\textsuperscript{-24} and random unidirectional p=value=8.2x10\textsuperscript{-26}, respectively).} 
 \label{fig:precursors}
\end{center}
\end{figure}

Furthermore, the basic properties of primary piRNA precursors were identified; presence of a poly(A) tail, their coding potential and expression levels. Poly(A) tails on the $3\textprime$ end of a transcript stabilise the RNA and prolong its life span \citep[review in][]{Garneau:2007kq}. Transcripts without the poly(A) tail are usually directed to the perinuclear granules, such as the chromatoid body in adult testis, where they are degraded by a set of exo- and endonucleases \citep{Balagopal:2009gda}. The primary piRNA transcripts  were either not polyadenylated, or their levels were lower than those of protein-coding mRNA (Figure \ref{fig:precursors}, A). Furthermore, a test was performed whether piRNA precursors can code for any proteins. The coding potential was calculated for all transcripts (Methods, Section \ref{sec:cpc}), and it was significantly smaller than for the protein-coding genes (Figure \ref{fig:precursors}, B). 

	\subsection{Epigenetic landscape of primary piRNA transcripts}


Having found that the piRNA cluster loci are mainly RNAP II transcribed, the epigenetic characteristics of the other protein-coding genes were investigated. Thus, chromatin immunoprecipitation (Methods, Section \ref{sec:ChIPseq.libraries}) were performed on three major histone marks: H3K4me3, H3K36me3 and H3K27me3. 

\begin{figure}[hbtp!]
\begin{center}
\includegraphics[width=14cm]{expression_correlation}
\caption[Correlation of ChIP-seq results  with tissue specific gene expression.]{Correlation of ChIP-seq results with tissue specific gene expression for protein-coding genes. X-axis denotes the rank of protein-coding genes ordered by the fold change of expression in testis versus liver. The genes on the right are highly expressed in testis and lowly in liver. Y-axis denotes rank of the genes ordered by the fold change of sequence counts in testis versus liver. The genes on top are highly enriched for specific histone marks or RNAP II. Relative levels of protein-coding gene expression between testis and liver (RPKM $>$1) were correlated with relative levels of DNA binding enrichment of H3K4me3, H3K26me3, H3K27me3 and RNAP II, with Spearman correlation coefficient of 0.767, 0.653, -0.636 and 0.770 respectively. Blue denotes the number of genes per hexagon, with darker blue representing a higher density of genes. The red curve denotes trends in data location with locally weighted scatterplot smoothing (LOESS). } 
 \label{fig:expression_correlation}
\end{center}
\end{figure}


\begin{figure}[hbtp!]
\begin{center}
\includegraphics[width=14cm]{epigenetic_landscape}
\caption[Epigenetic landscape of piRNA precursor genomic loci.]{Epigenetic landscape of piRNA precursor genomic loci. A. Mature piRNAs (yellow) and long non-coding RNA transcripts (red) (plus, sense; minus, antisense) were mapped to a typical bidirectional piRNA gene locus in testis and liver. For the same locus, the enrichment of RNAP II (Pol II, purple), H3K4me3 (grey), H3K36me3 (blue), and H3K27me3 (green) was determined. The y-axis represents normalised read density for RNA-seq data or read count for ChIP-seq data, while the x-axis indicates chromosomal location. B. Differential enrichment versus input of all the ChIP proteins over annotated gene loci in testis and liver. Significance is indicated by circle radius (high: large; low: small). Log (base 2) fold-changes over input are shown from blue (negative) to blue (positive).} 
 \label{fig:epigenetic_landscape}
\end{center}
\end{figure}

As a functional control experiment, the behaviour of the histone mark enrichment in relation to the expression of protein-coding genes had to be confirmed. The H3K4me3 and H3K36me3 histone marks are enriched at the sites of active transcription \citep{SantosRosa:2002p598,KolasinskaZwierz:2009p603}, while H3K27me3 is enriched in silenced genomic loci \citep{Ringrose:2004p604}. It was tested whether the difference in binding enrichment between testis and liver tissues for these marks correlates with the respective difference in gene expression. For example, genes that are highly expressed in testis, but lowly in liver, should have the same pattern for the H3K4me3 binding enrichment and opposite for H3K27me3. To that end, a differential expression analysis was performed on proteins that are expressed in both testis and liver (RPKM $>$1) (Methods, Section \ref{sec:diffexprs}). This provided a ranked list of genes that were ordered according to their tissue specific expression. Second, the same analysis was then performed for each of the histone marks, H3K4me3, H3K36me3, H3K27me3 as well as for RNA polymerase II. Similarly to the analysis of gene expression levels, this resulted in a list of genes ordered by their tissue specific binding enrichment. As expected, the tissue specific gene expression was correlated with the marks of active transcription H3K4me3 and H3K36me3 as well as RNAP II, but anti-correlated with the repressive mark H3K27me3 (Methods, Section \ref{sec:epigenetic_marks}) (Figure \ref{fig:expression_correlation}).

Furthermore, the patterns of each of the histone modifications were observed over the piRNA precursor loci. Transcription start sites of the piRNA transcripts were occupied by H3K4me3, while H3K36me3 was more pronounced at the gene bodies (Figure \ref{fig:epigenetic_landscape} A). H3K27me3 as a mark of facultative heterochromatin \citep{Mikkelsen:2007jg} was not observed at the genomic regions producing piRNAs, neither at the body of the piRNA genes for liver, or at the promoter sites for testis. This suggested that these loci are not repressed via mechanisms involving the formation and maintenance of heterochromatin during terminal differentiation in somatic tissues.  Furthermore, the enrichment pattern of the selected marks over the input showed the greatest similarity to the other RNAP II dependent transcripts (Figure \ref{fig:epigenetic_landscape} B). 

Overall, the majority of the primary piRNA transcripts had epigenetic properties similar to the RNA polymerase II transcribed protein-coding genes. 



	\subsection{Transcription regulation of primary piRNA transcripts}

Although transcriptional regulation can occur at distance, most transcription activity is controlled by cis-acting regulatory elements bound by tissue-specific master regulators \citep{Romero:2012hu}. The previous results on genomic location of piRNA precursors allowed us to identify the sites that most likely serve as proximal promoter regions of piRNA genes. Information contained in these sites can be used to discover patterns of transcription regulation by cis-acting elements.  

	In order to investigate the transcriptional regulation of piRNA precursors, their promoter sequences were extracted and motif search was performed (Methods \ref{sec:myb.identification}). The enrichment of the binding site motifs was observed for transcription factors SP1 (p-value 2.5x10\textsuperscript{-5}) and PAX4 (p-value 2x10\textsuperscript{-4}), that seem to recognise similar sequences \citep{Butler:2002p636}. A binding motif for the transcription factor myeloblastosis oncogene-like 1 (MYBL1), also termed A-MYB, was significantly overrepresented with a p-value of 3.6x10\textsuperscript{-5} (Figure \ref{fig:MybL1_motif_identification}, A). MYBL1 is a testis specific transcription factor (TF) that regulates a large proportion of spermatogenesis genes \citep{BolcunFilas:2011p637}. Knockouts of this TF are known to cause male sterility and stall of the spermatogenesis at the pachytene stage  \citep{Toscani:1997p638}, similar to Mili \citep{KuramochiMiyagawa:2004eu} and Miwi \citep{Deng:2002wb} knockouts. The MYBL1 motif was enriched at the promoter sites of putative piRNA primary transcripts and was at the level observed at the known MYBL1-regulated protein-coding genes, as predicted by ChIP-chip experiments (Figure \ref{fig:MybL1_motif_identification}, B). 
	To further verify the previous observations, a ChIP experiment was performed with MYBL1 specific antibody. The enrichment of MYBL1 binding was present at 84.3\% bi- and 47.2\% unidirectional-associated piRNAs and 8.9\% of testis expressed proteins. When  the top 30 bidirectional and unidirectional piRNA clusters were aligned by the MYBL1 peaks closest to their promoter site, they exhibited the same epigenetic pattern as the MYBL1-associated protein-coding genes (Figure \ref{fig:heatmap}). In contrast, MYBL1 was not expressed in liver, nor did similar ChIP-seq experiments reveal any genomic occupancy. This observation confirmed that  piRNA transcripts are regulated in a similar fashion as the other protein-coding genes, with features including both the epigenetic hallmarks of transcription and proximal promoters enriched for cis-acting motifs that are directly bound by germ-line transcription regulators.


\begin{figure}[hbtp!]
\begin{center}
\includegraphics[width=14cm]{MybL1_motif_identification}
\caption[Enrichment of MYBL1 motif in promoters of piRNA precursors]{Enrichment of MYBL1 motif in the promoter sequences of piRNA precursors. A. A motif finder tool MEME was used to identify sequences that are enriched in the promoters of the primary piRNA transcripts. The top row marks the canonical 8mer, while a 21 nucleotide long motif was identified from the piRNA precursor sequences. The y-axis represents the information content in bits. B. Proportion of sequences containing a MYBL1 motif in relation to the distance from a promoter site. The y-axis represents the ratio of promoter sequences with the MYBL1 motif. The piRNA precursor gene loci (red) are compared to known MYBL1-regulated protein-coding genes (yellow) \citep{BolcunFilas:2011p637}, MYBL1 independently regulated protein-coding genes (green), and all genes (blue).} 
 \label{fig:MybL1_motif_identification}
\end{center}
\end{figure}
	


\begin{landscape}
\begin{figure}[hbtp!]
\begin{center}
\includegraphics[width=22.5cm]{heatmap}
\caption[Binding of MYBL1 to promoters of piRNA precursors]{Binding of MYBL1 to promoters of the piRNA precursors. Top 30 bidirectional (divergent arrows) and unidirectional (single arrow) expressed loci are presented in the following categories: piRNA genes, protein-coding genes whose promoter is bound by MYBL1, randomly selected genes. Heatmaps are shown for total RNA and piRNAs, as well as occupancy of RNA polymerase II (Pol II), active histone marks H3K4me3 and H3K36me3, repressive histone mark H3K27me3 and MYBL1 (blue: presence; white: absence); all experiments are shown for the same loci in mouse testis and liver.
} 
 \label{fig:heatmap}
\end{center}
\end{figure}
\end{landscape}

%\InsertFig{\IncludeGraphicsH{heatmap}{6cm}{}}{Epigenetic landscape of the MYBL1 peak aligned piRNA regions, in relation to the MYBL1 protein-coding genes and random expressed regions.}{fig:heatmap} 
%
	\subsection{Production of piRNAs from convergent promoters}

In \textit{D. melanogaster}, piRNAs can originate from the $3\textprime$ UTRs of convergently transcribed genes \citep{Robine:2009hy}. This observation suggested a hypothesis that the mammalian piRNA processing pathway might require two long overlapping RNAs transcribed from opposite DNA strands. This pathway had to be restricted to germ-cells because they are immunologically privileged and avoid the normal interferon response caused by dsRNAs \citep{Dejucq:1998tt}.

While inspecting the piRNA cluster loci, it was noticed that each bidirectional cluster contained several H3K4me3 non-promoter peaks (Figure \ref{fig:ricochet} A). Strand specificity of the total RNA and piRNA reads prompted us to hypothesise that a transcription in the opposite direction from the main piRNA cluster starts at several sites, but at a much lower intensity (Figure \ref{fig:ricochet}). The pattern was observed for the bidirectional sites, with a median of four promoter regions enriched for H3K4me3 and ranging approximately 70kb in size (Figure \ref{fig:ricochet}, Table \ref{tab:H3K4counts}, Table \ref{tab:H3K4width}).


\begin{figure}[hbtp!]
\begin{center}
\includegraphics[width=14cm]{ricochet}
\caption[Convergent transcription gives rise to piRNA clusters]{Convergent transcription gives rise to piRNA clusters. A. An example of piRNA production from the left wing of a piRNA cluster on chromosome 17. Grey arrows indicate directionality and strength of transcription. Sequence coverage of total RNA, piRNAs, H3K4me4 and MYBL1 is shown. B-D. Examples of genome tracks for four types of piRNA loci with convergent transcription. B. First row: bidirectional (Mmus\_bi\_15) with annotated reads of piRNAs (yellow), total RNA (red) and H3K4me3 (grey). Second row:  unidirectional type I (Mmus\_uni\_16) with two non-coding RNAs. Third row:  unidirectional type II  (Mmus\_uni\_49) formed by a non-coding and a protein-coding transcript (blue). Bottom row:  unidirectional type III  (Mmus\_uni\_60) formed by two protein-coding genes. C. Schematics of the four classes. D. Histogram of widths for each type.} 
 \label{fig:ricochet}
\end{center}
\end{figure}


% Table generated by Excel2LaTeX from sheet 'Sheet1'
\begin{table}[htbp]
  \centering
  \caption[Number of H3K4me3 peaks in piRNA producing regions]{The number of regions enriched for H3K4me3 near piRNA loci. Summary of total number of H3K4me3 peaks over bi- and unidirectional piRNA loci, testis-expressed protein-coding genes and a sub-fraction of those, which are bound by the MYBL1 transcription factor. Columns represent the mean, standard deviation (sd)\nomenclature{sd}{standard deviation} and median absolute deviation (MAD)\nomenclature{MAD}{median absolute deviation}.
}
    \begin{tabular}{lrrrr}
    \toprule
    Genomic region & \multicolumn{4}{c}{Number of H3K4me3 peaks per region} \\
    \midrule
    & mean  & sd    & median & MAD \\
    Bidrectional & 4.47  & 2.78  & 4     & 2.97 \\
    Unidirectional type I & 2.36  & 0.96  & 2     & 0.00 \\
    Unidirectional type II & 2.84  & 1.49  & 2     & 1.48 \\
    Unidirectional type III & 2.60  & 1.07  & 3     & 1.48 \\
    Protein-coding genes & 1.60  & 1.65  & 1     & 0.00 \\
    MYBL1-bound protein-coding genes & 2.63  & 3.88  & 2     & 1.48 \\
    \bottomrule
    \end{tabular}%
  \label{tab:H3K4counts}%
\end{table}

% Table generated by Excel2LaTeX from sheet 'Sheet$2\textprime$
\begin{table}[htbp]
  \centering
\caption[Width of piRNA producing regions]{The width of regions between H3K4me3 near piRNA loci. Columns represent the mean, standard deviation (sd) and median absolute deviation (MAD).
}
    \begin{tabular}{lrrrr}
    \toprule
          & \multicolumn{4}{c}{Width of regions between H3K4me3 peaks} \\
    \midrule
    Type  & mean  & sd    & median & MAD \\
    Bidrectional & 80502 & 48751 & 67156 & 52712 \\
    Unidirectional type I & 29941 & 18925 & 28805 & 15343 \\
    Unidirectional type II & 49185 & 27007 & 47460 & 35641 \\
    Unidirectional type III & 72927 & 79175 & 52111 & 34506 \\
     \bottomrule
    \end{tabular}%
  \label{tab:H3K4width}%
\end{table}% 


A related architecture was observed for intergenic piRNA loci, whose genomic structure resembled an isolated, unidirectional wing of a bidirectional cluster (unidirectional type I, Figure \ref{fig:ricochet}). A strong, prominent peak of H3K4me3 initiated primary long noncoding RNA transcription, which converged on a complementary noncoding RNA transcribed at an often substantially lower level from a counterpart transcriptional initiation site. These intergenic, unidirectional clusters had a median of two regions enriched for H3K4me3 and were circa 30 kb long (Table \ref{tab:H3K4counts}, Table \ref{tab:H3K4width}). 


Combining the mapping of total RNA and piRNA with transcriptional initiation marked by H3K4me3 in vivo, provided mechanistic insight into the recent observation that piRNAs can arise from $3\textprime$ UTRs of protein-coding genes \citep{Robine:2009hy}. The third class of piRNA clusters that were discovered were driven by a strong transcription initiation site marked by H3K4me3 whose product RNA was antisense to the direction of the protein-coding gene transcription, and thus could include the $3\textprime$ UTR of protein-coding genes (unidirectional type II, Figure \ref{fig:ricochet}). In addition, the fourth class of piRNAs was generated from the DNA sequences located between two adjacent genes that are transcribed convergently (unidirectional type III, Figure \ref{fig:ricochet}). 


In order to test the hypothesis of necessity of sense-antisense convergent transcription for the production of piRNAs,  both the sense and the antisense transcripts from the same loci were quantified (Methods, Section \ref{sec:RTqPCR}). Strand specific probes for the quantitative PCR were designed to confirm that the transcripts in both directions are present. Both the sense and the antisense primary piRNA transcripts were detected, and the antisense one was expressed at a lower intensity (Figure \ref{fig:RTqPCR}).
 
In summary, despite having four distinct architectures, all piRNA clusters in mouse testis shared certain features. First, these clusters had highly expressed, noncoding RNAs that have been predicted to mature into piRNA molecules \citep{Aravin:2006p384,Vourekas:2012p607}. Second, the transcription initiation sites flanking the piRNA clusters were identified, as well as the antisense RNAs they produced, though often at a lower level (Figure \ref{fig:RTqPCR}). Taken together, the transcriptional architecture of all piRNA loci explains how the action of two opposing DNA-dependent RNA polymerases can create transcripts of opposite strandedness for further processing into piRNAs.

\begin{figure}[hbtp!]
\begin{center}
\includegraphics[width=14cm]{qPCR}
\caption[Validation of antisense gene expression with RT-qPCR]{Validation of antisense gene expression convergent to piRNA precursor transcripts. Transcript abundance of selected bidirectional (left) and unidirectional (right) piRNA precursor transcripts was validated by strand-specific RT-qPCR. Here, the sense (s) transcript is stronger expressed than the antisense (as) transcript. The expression values were normalised against Actin B. Error bars represent means ± SE of the mean (n=3). Figure adopted from Claudia Kutter, with permission. } 
 \label{fig:RTqPCR}
\end{center}
\end{figure}
	

\subsection{Formation of piRNA producing loci during evolution}

The existence of piRNA clusters originating from convergent transcription of two protein-coding genes suggested a possible mechanism of their birth during evolution. The most common changes in the genome are local rearrangements where a segment of DNA is inverted in orientation relative to a common ancestor \citep{Zhao:2009gb}. 

	Since previous interspecies mapping had not discovered examples of this mechanism in rat \citep{Lau:2006ka}, human \citep{Girard:2006gu} or platypus \citep{Murchison:2008kp}, piRNA clusters were mapped in testes of \textit{Canis familiaris} (dog) and \textit{Monodelphis domestica} (opossum). \textit{C. familiaris} was chosen as a placental mammal that diverged approximately 85 million years (MY) from mouse, which has had substantially accelerated karyotypic rearrangement compared to the above eutherians \citep{LindbladToh:2005kr}.  \textit{M. domestica} was chosen as an evolutionarily distant marsupial mammal (180 MY from mouse) \citep{Mikkelsen:2007jg}. 

Similar to mouse, sequences from  H3K4me3 ChIP-seq and total RNA-seq were analysed to map the transcription initiation sites and precursor RNAs genome-wide in testis of  \textit{C. familiaris} and \textit{M. domestica} (Methods, Sections \ref{sec:ChIPseq.libraries} and \ref{sec:RNA.seq.libraries}). In addition, small RNAs were isolated and sequenced in order to capture the majority of mature piRNA species. This afforded 97 and 274 piRNA clusters in  \textit{C. familiaris} (Appendix, Table \ref{tab:dogClusters}) and \textit{M. domestica} (Appendix, Table \ref{tab:opossumClusters}) respectively, consistent with prior reports in rat and human \citep{Lau:2006ka,Girard:2006gu}. Of the 178 murine piRNA loci, 81 (45.5\%) in  \textit{C. familiaris} and 12 (6.7\%) in \textit{M. domestica} were orthologous and expressed (Appendix, Table \ref{tab:mouseConservation}). 

\begin{figure}[hbtp!]
\begin{center}
\includegraphics[width=14cm]{inversion}
\caption[Gene inversion produces a piRNA cluster]{An opossum-specific inversion of a genomic region containing two genes has produced a unidirectional type III piRNA cluster not present in eutherian mammals. Evolutionary tree illustrating lineage divergence of mouse (Mmus), dog (Cfam) and opossum (Mdom). A mouse genomic region (top) encoding protein-coding genes Chuk, Cwf19l1, Bloc1s2 and Pkd2l1 on the minus strand (blue arrows) is conserved by synteny and gene content in dog (middle). In opossum (bottom), the gene content is preserved, but a regional inversion has placed the Bloc1s2 and Cwf19l1 genes onto the plus strand. For each species, tracks with annotated reads piRNAs (yellow), total RNA (red) (positive, sense; minus, antisense) and H3K4me3 (grey) are shown. The y-axis specifies normalised read density. The cluster of piRNAs is only present in opossum. 
} 
 \label{fig:inversion}
\end{center}
\end{figure}

In  \textit{C. familiaris} and \textit{M. domestica}, a total of 28 piRNA clusters were identified that were created when two genes were arranged tail-to-tail in the genome (unidirectional type III). Out of 390 and 203 syntenic convergent regions, 5.6\% and 2.0\% were found with altered directionality in  \textit{C. familiaris} and \textit{M. domestica}, respectively. All unidirectional type III piRNA loci in mouse were syntenically conserved in  \textit{C. familiaris} and therefore were not newborn clusters. In \textit{M. domestica}, a significant fraction of these unidirectional type III clusters were syntenic to mouse, yet one of the remaining loci on chromosome 1 was flanked by a genomic segment containing two protein-coding genes whose orientation had been inverted relative to the common ancestor of eutherian mammals (Figure \ref{fig:inversion}). Remarkably, this locus showed the previously described hallmarks of piRNA production, including H3K4me3 enrichment at the TSS of the flanking protein-coding genes, potential long noncoding RNA precursors identified from the total RNA mapping, and mature piRNA sequences oriented in the same direction as the potential precursors. This locus in the \textit{M. domestica} genome is therefore an unambiguous example of how a small genomic rearrangement containing protein-coding genes can also result in \textit{de novo} creation of a novel piRNA cluster.

	

\section{Conclusion}

The main steps in the  biogenesis of mammalian pachytene piRNAs are slowly being identified. However, little is known of the first steps in pachytene piRNA biogenesis involving transcription from their genomic loci. In this work, a set of experiments were performed to investigating transcriptional and epigenetic properties of these loci throughout evolution. 

The total RNA sequencing permitted the identification of the long noncoding RNA thought to serve as the precursor for piRNA maturation. These primary transcripts were combined with genomic locations of piRNAs and transcriptional initiation sites revealed by H3K4me3 enrichment to further define the transcriptional units responsible for piRNA production. Furthermore, little has been known regarding the transcriptional regulatory mechanisms driving the expression of piRNA precursors. This work shows that these piRNA cluster loci have hallmarks of other genes transcribed by RNA polymerase II, including enrichment of H3K4me3 at the TSSs\nomenclature{TSS}{transcription start site}, enrichment of H3K36me3 in regions of transcriptional elongation, and direct promoter binding by a transcription factor, identified as MYBL1.

Further analysis of the transcriptional units revealed four major architectures in the mammalian genome that produce piRNAs. All architectures create convergent transcription over a region of approximately 50 kb. One strand is almost always dominant, with a weaker transcript originating from the second strand, suggesting that biogenesis of piRNAs requires overlapping antisense transcripts. Like all members of the AGO protein family, mouse PIWI proteins may not distinguish between plus and minus strand, but may have strand-preference for the more highly expressed transcript. These antisense transcripts can originate from protein-coding genes, noncoding RNA loci, or their combination. Furthermore, such architecture also points to the possible solution of piRNA amplification in the absence of transposable elements. During the biogenesis of pre-pachytene piRNAs, transposon elements provide the material for the amplification during the ``ping pong'' cycle. Existence of piRNA precursors from convergent loci ensures that both strands are available for the amplification of pachytene piRNAs. 

These observations allow for several explanations for the formation of the unidirectional piRNA clusters. First, the appropriate antisense transcript could be missing from one side of a bidirectional promoter, despite the existence of bidirectional transcription. Second, MYBL1 might be producing bidirectional transcripts of unequal intensity, whereas the expression of the lowly expressed transcript is insufficient to be detected with the current sequencing methods. Finally, parts of the protein-coding transcripts that are not protected by the translation machinery can result in a production of piRNAs if they have a converging antisense transcript, whether it is intergenic or protein coding. This would explain the enrichment of piRNAs coming from the $3\textprime$ UTR regions of protein-coding genes, previously observed in both \textit{D. melanogaster} and mammals \citep{Robine:2009hy}. 

The confirmation of the evolutionary conservation of piRNA-forming architectures as well as the mechanism of their evolutionary formation came from the analysis of pachytene piRNA loci of \textit{C. familiaris} and \textit{M. domestica}. These loci were tested for their total RNA production, piRNA formation and characteristics of epigenetic marks. It was demonstrated that more than half of piRNA loci are diverged between mouse and \textit{C. familiaris} and almost all for \textit{M. domestica}. Furthermore, the analysis of the structure of these piRNA precursor loci allowed for the deduction of the likely mechanism by which new piRNA clusters are formed through genomic inversions. Inversions are the most common genome rearrangements during evolution. Such rearrangements can result in convergent transcription from two TSSs that were originally oriented on a same strand. This mechanism would provide a steady state maintenance of piRNA production, given that there is continual turnover of piRNA loci and the new ones have to be constantly formed.

The strategy of exploring the pachytene piRNA loci  in testes from three species has confirmed many hypotheses regarding the location and production of these non-coding RNAs and extended our understand of their transcriptional and regulatory architectures. 


\newpage
\section{Methods}
\label{sec:Methods_piRNA_transcriptomics}

	\subsection{Computational biology methods}
	\subsubsection{Preparation of piRNA libraries} 
	\label{sec:GEO.piRNA.libraries}

Piwi protein interacting RNA reads from Mili-IP (GSM475280) and Miwi-IP (GSM475279) were downloaded from GEO, record GSE19172 \citep{Robine:2009hy}. The reads were read into R using the library RSamTools and the function scanBam (v.1.2.3.). The reads from Mili and Miwi immunoprecipitations were combined into one R object which represented total piRNA in the adult testis. 

	\subsubsection{Cleaning and filtering of sequences}
	\label{sec:cleaning.filtering}
The reads in fast format were submitted to cleaning and filtering with Reaper (van Dongen et al. in review, v.12-103), available from \url{
http://www.ebi.ac.uk/~ stijn/ reaper/src/reaper-12-103.tgz}. The smallest amount of unmapped reads was received with the median quality of 50 in a window of 9, with the minimum length of cutting after 20 base pairs of the read. The minimum length of a read was 16. The reads were cleaned of the base pairs with multiple B quality scores, assigned by the Illumina machines when encountering a deteriorating quality in a read. The downstream parts of reads were discarded if 3 out of 5 base pairs encountered were read as N, when no nucleotide was called. The redundant reads were thereafter collapsed with an in-house developed tool Tally (van Dongen et al. in review) v.12-103, available from the same page as Reaper \url{http://www.ebi.ac.uk/~stijn/ reaper/src/reaper-12-103.tgz} as well as Reaper. The same procedure was conducted for the files consisting of multiple technical replicates of the same sample, combined at the level of fastQ files. 

	\subsubsection{Mapping to the genome}
	\label{sec:genome.mapping.new}
The reads were mapped to the mouse, \textit{C. familiaris} genome (Canis.familiaris. BROADD2) and \textit{M. domestica} genome Monodelphis.domestica.BROADO5 with Ensembl version 67 \citep{Flicek:2011p677}. The mapping was performed with Bowtie 0.12.7 allowing for two mismatches \citep{Langmead:2009fv}. All sequence reads that map to the genome were reported, and the reads that mapped to more than one location with equal quality were randomly allocated as suggested in \citep{Treangen:2012p679}. The other Bowtie options included ``-best --strata''for reporting an alignment for a read with the smallest numbers of mismatches. The reads were reported as files in the SAM format and converted to BAM format. The bam files were indexed, the reads counted using samtools and the results plotted. 

	\subsubsection{Genome annotation}
	\label{sec:genome.annotation}
The genome annotation was downloaded with R (v.2.12) package biomaRt (v.2.6) for Ensembl annotation (v.65) \citep{Flicek:2011p677}. Pseudogenes and nonsense mediated decay products were removed from the annotation, and reported as ``other''. The regions were imported into R objects with the library Genomic Ranges (v.1.2.3) and the overlaps and redundancies were removed with the function reduce. The regions that mapped to several categories were assigned only to one with the following hierarchy: ``piRNA loci'', ``protein coding'', ``miRNAs'', ``snRNAs'', ``rRNA 5.8S, 18S, 28S'', ``rRNA 5S'' , ``tRNAs'', ``LINEs'', ``SINEs'' and ``LTRs''. The protein-coding regions were confined only to the ones which were expressed at the tissue at the time, as reported by the Cufflinks with RPKM of at least 1. 

	\subsubsection{Removal of annotated regions from the piRNA clusters}
	\label{sec:cluster.cleaning}
In order to focus on intergenic, non-repeat associated clusters, the reads were cleaned from annotated regions in the following steps. The coverage of the reads was calculated in a strand specific manner. Regions with the coverage of at least 50 in the window size of 1000 were reported, due to the largest number of piRNA retained (Figure \ref{fig:cluster_cutoffs}). The information from the positive and the negative strand was combined. Granularity of the regions was decreased by combining the clusters closer than 10kb. To that end, the regions were first expanded by 10kb on each side and then combined with the function reduce. The clusters were then trimmed to the minimum start site and the maximum end site of the combined regions. For each of the categories: non-coding RNAs, protein coding, other and repeats - reads that mapped to that region were discarded and the clusters were called again with the same criteria. The final list of clusters was annotated with those that overlap with an H3K4me3 peak as a mark of active transcription (see \ref{sec:peak.calling}). The clusters which overlapped after the strand directionality was removed were annotated as bidirectional. 


\begin{figure}[hbtp!]
\begin{center}
\includegraphics[width=14cm]{cluster_cutoffs}
\caption[Optimisation of parameters for calling piRNA clusters.]{Optimisation of parameters for calling piRNA clusters. The piRNA clusters were called with different numbers of piRNA allowed per window of 1000 nucleotides. The number of piRNAs ranged from 5 to 300 with steps of 5. Besides calculating the number of piRNAs that get assigned to a clusters, the number of clusters, as well as their width and average piRNA content was calculated.} 
 \label{fig:cluster_cutoffs}
\end{center}
\end{figure}

	\subsubsection{Plotting genome annotation}
	\label{sec:genome.annotation}
The reads were extended to the estimated fragment length of 200bp and overlapped with the known genomic regions, acquired in \ref{sec:genome.annotation}. The counts of the overlapping reads per annotation region were used to calculate differential expression versus the input as described in Section \ref{sec:diffexprs}. The library ggplot2 version 0.8.9 was used to plot the samples. The p-values were separated into three categories, ``$>$0.0$5\textprime$' as insignificant, ``$<$0.05 and $>$0.01'' as marginally significant and ``$<$0.01'' as significant. 

\subsubsection{Calculation of differential expression}
	\label{sec:diffexprs}
	The differential expression calculations were performed with the Bioconductor package DEseq, (v.1.2.1) \citep{Anders:2010fu}. Instead of the estimated total number of counts per library, the counts of the number of mapped reads were used. The variance was estimated based on the pooled samples. The fold change and the multiple testing corrected p-values were calculated with the default parameters. 

	\subsubsection{Transcriptional landscape}
	\label{sec:epigenetic_marks}
Several samples of bidirectional and unidirectional cluster region were chosen as examples of the cluster's transcriptional landscape. Coverage of the reads that overlap the regions was taken from the adult testis piRNA libraries, RNA-seq libraries and combined ChIP-seq libraries for H3K4me3, H3K36me3 and H3K27me3. For the stranded libraries (piRNAs and RNA-seq) the coverage was taken per strand, and the density was plotted, weighted by the number of reads on each strand. Library ggplot2 (v.0.8.9) was used to visualise the density, using the function geom\_density, with the alpha parameter of 0.7. For the non-stranded libraries the coverage counts were used. Smoothing was performed with the function stat\_smooth, for the linear model with natural cubic splines with 150 knots. 

	\subsubsection{Global patterning of polymerases and histone marks}
In order to assess the global patterns of piRNA cluster transcription, it was important to visualise all the libraries at once, for a subset of piRNA clusters and control regions. To that end, a set of 30 bidirectional and unidirectional transcription loci from the following categories was chosen: piRNA clusters, protein-coding genes with MYBL1 associated transcription, and random testis expressed protein-coding genes. The regions were selected in the following manner. First, for the piRNA clusters, the start sites of the clusters were identified, and adjusted to the summits of the MYBL1 peaks if they were present in the area around 10kb on both sides of the start site. If the cluster start sites did not overlap any of the MYBL1 peaks, the clusters were either centred in between the start sites of the two bidirectional transcripts, or left as is if unidirectional. The clusters were sorted by the total number of piRNAs in the cluster. Only the region of 10kb around the start sites was selected. 
For the protein-coding genes, transcripts which contained a MYBL1 peak were selected and ordered by the FPKM value, as calculated by Cufflinks \citep{Trapnell:2010p681}. Two genes were called as bidirectional if they were on the opposite strand and their start sites overlapped when extended by 10kb. As for the piRNA clusters, the regions were adjusted to the MYBL1 peak summits if in the 10kb vicinity of the locus start site. Random protein-coding genes which did not contain the MYBL1 peak were selected as a control.

	\subsubsection{Peak calling}
	\label{sec:peak.calling}
The peak calling was performed with Macs (v.1.4.1) with the default parameters on the libraries combined by the immunoprecipitated protein. The parameters for the calling of the MYBL1 peaks were pval=0.01 and mfold=3,30 to increase the sensitivity. The peak calling was initially performed also with CCAT with the default parameters for wide peaks \citep{Xu:2010p682}. However, the number of peaks was smaller or equal to Macs, so Macs was chosen as the default peak caller for simplicity. 

	\subsubsection{MYBL1 identification}
	\label{sec:myb.identification}
The regions at the presumed transcription start site of the bidirectional clusters were identified individually for the top 50 bidirectional clusters with the genome broser IGV (v.2.0.7). The sequences were extracted with R, packages BSgenome and BSgenome.Mmusculus. UCSC.mm9 (v.1.18.3). The reads were submitted to the Meme-Chip online service (v.2.0.7). MYBL1 was identified as a candidate based on the tissue specificity and similarity of phenotype to Mili knockout. The MYBL1 motif was searched in the promoter regions of piRNA loci, known MYBL1 regulated genes and all protein genes. Promoter regions of 100 to 1500bp were selected and a percentage of total promoters with at least one MYBL1 motif identified by Fimo (Meme package, v.4.6.1) was reported, with the default settings and the p-value cut-off of 5e-4. 

	\subsubsection{Transcriptome assembly}
	\label{sec:cufflinks}
Raw reads in fastQ format from the paired end, strand specific sequencing were submitted to GSNAP (v.2012-01-11). Antisense was expected on the first read of the pair, while sense on the second, and the quality protocol was ``sanger''. The mapping was performed with 2 allowed mismatches, distant splice penalty of 100, expected distance between the pairs of 160 based on the average fragment length, with search for novel splicing and without trimming of the reads at the ends. The report contained a maximum of one transcript and no failed alignments. The files were separated into 200 pieces, and each was parallelised over 2 processes on the EBI computer cluster. The bam files were combined, sorted and indexed. The mapped reads were submitted to cufflinks with the default parameters \citep{Trapnell:2010p681}. 

	\subsubsection{Calculation of coding potential}
	\label{sec:cpc}
The coding potential calculation was performed with CPC tool, with default parameters \citep{Kong:2007hx}. The analysis was performed for 100 testis-expressed protein-coding genes, bi- and unidirectional piRNA precursor transcripts identified in this study, as well as randomly assembled sequences of bi- and unidirectional piRNA precursor transcripts. 

	\subsection{Experimental methods}
	\subsubsection{Tissue preparation}
	\label{sec:tissue.preparation}
Experiments were performed on testis and liver material isolated from mouse (Mus musculus C57BL/6), dog ( \textit{C. familiaris}), and opossum (\textit{M. domestica}) on at least two independent biological replicates from different animals, except for RNA-seq for dog (\textit{Canis familiaris}) and opossum (\textit{Monodelphis domestic}). Mice (males, 2 month old) were obtained from the Cambridge Research Institute under Home Office license PPL 80/2197.  \textit{C. familiaris} testis tissue (male; 14 months old) was obtained from Harlan (UK) and \textit{M. domestica} testis tissue (male, 17 month old) from the University of Glasgow (Glasgow, UK). All tissues were treated post-mortem with 1\% formaldehyde for ChIP-seq or flash-frozen in liquid N2 for RNA experiments.

	\subsubsection{ChIP-seq library preparation}
	\label{sec:ChIPseq.libraries}
	
After tissue preparation (Section \ref{sec:tissue.preparation}) ChIPseq experiments were performed. The DNA was acquired as described in Schmidt et al. 2009 \citep{Schmidt:2009ca} using the following antibodies H3K4me3 (Milipore, CMA304), H3K36me3 (abcam, ab9050), H3K27me3 (E1F) \citep{Chandra:2012p684}, Pol I (sc9131 RPA194), Pol II (abcam, ab5408), Pol III (1900). The immunoprecipitated DNA was end-repaired, A-tailed, ligated to the sequencing adapters, amplified by 18 cycles of PCR and size selected (200-300 bp). Single end adapters (Illumina) were ligated to the fragmented products and 
PCR amplified \citep{Parkhomchuk:2009p685}. After passing quality control on a Bioanalyzer 1000 DNA chip (Agilent) libraries were sequenced single ended on the Illumina Genome Analyzer II or HiSeq 2000 and post-processed using the standard GA pipeline software v.1.4 (Illumina). 

	\subsubsection{RNA-seq library preparation}
	\label{sec:RNA.seq.libraries}
	
The tissues were prepared as explained previously (Section \ref{sec:tissue.preparation}). For RNA-sequencing experiments, total RNA was extracted using Qiazol reagents (Qiagen) and DNase-treated (Turbo DNase, Ambion). Ribosomal RNA was depleted from total RNA using RiboZero (Epicentre). RNA was reversed transcribed and converted into double-stranded cDNA (SuperScript cDNA synthesis kit, Invitrogen), chemically fragmented. Paired end adapters (Illumina) were  ligated to the fragmented products and prior to PCR amplification cDNA was UNG-treated to maintain strand-specificity \citep{Parkhomchuk:2009p685}. After passing quality control on a Bioanalyzer 1000 DNA chip (Agilent) libraries were sequenced on the Illumina Genome Analyzer II or HiSeq 2000 (ChIPseq libraries single-ended, RNA-seq libraries paired-ended) and post-processed using the standard GA pipeline software v1.4 (Illumina). 

\subsubsection{Reverse transcription followed by quantitative PCR}
	\label{sec:RTqPCR}
The reverse transcription of selected piRNA gene loci was performed using \SI{5}{\micro\gram} of DNase-treated total RNA according to the manufacturer's protocols using 200 U SuperScript II Reverse Transcriptase (Invitrogen Corporation), 150 ng random primers or 2 pmol of gene-specific RT primers. Negative controls were included in RT reactions. The cDNAs were then treated with RNase H at 37$^\circ$C for 20 min. Each PCR reaction typically contained 5 pmol of the gene-specific primers, \SI{10}{\micro\liter} Power Sybr (Agilent), and \SI{2}{\micro\liter} of the diluted cDNAs in a total volume of \SI{20}{\micro\liter}. Reactions were carried out in triplicate in ABI 7900HT Fast Real-Time PCR system at the optimal temperature, as defined by provider instructions.

\subsubsection{Availability of sequencing data}
	\label{sec:arrayexpress}
The sequencing data has been deposited to ArrayExpress under accession number E-MTAB-1265. Additional sequencing data used in this study can be found at E-MTAB-424 (Pol III in liver and testis) and E-MTAB-941 (Pol II, H3K4me3 and H3K36me3 in liver). 


% ------------------------------------------------------------------------


%%% Local Variables: 
%%% mode: latex
%%% TeX-master: "../thesis"
%%% End: 


